### PR TITLE
ci: Set up release tagged job

### PR DESCRIPTION
**Pull request by francislavoie**
WIP

Added a job that updates the stackbrew config, probably a bit wordy but I think it does what we need there.

Still need to add the step to run your `make`, you'll need to let me know the best way to do that.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
@francislavoie why all the extra whitespace changes in the workflow files? can you revert those?

Also since it's a WIP, you can mark it as a draft - I'll do that now.

**Comment by francislavoie** [comment]
Readability. It's hard to tell apart each step from eachother without them IMO. I was trying to read it and audibly said "wtf is going on here" until I gave it some room. 🤷‍♂️

**Comment by hairyhenderson** [comment]
@francislavoie oh... it was pretty readable to me before 😉 - if it's important to you can you PR it separately? it's unrelated to this particular change.

File: .github/workflows/release-tagged.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          # TODO: Switch to event payload tag
+          TAG: v2.0.0-rc.3
+          # TAG: ${{ github.event.client_payload.tag }}
+        run: |
+          # Get the checksums file from the tagged release
+          wget https://github.com/caddyserver/caddy/releases/download/${TAG}/caddy_${TAG#v}_checksums.txt -O checksums.txt
+
+          # Update the version tag in the config
+          echo "Updating version to '${TAG#v}'"
+          sed -i "s/caddy_version: .*/caddy_version: ${TAG#v}/" stackbrew-config.yaml
```
Comment (hairyhenderson) [inline .github/workflows/release-tagged.yml:30]: using `yq` would probably be more reliable:

```suggestion
          yq write -i stackbrew-config.yaml caddy_version ${TAG#v}
```

(I presume `yq` is installed, if not you can run it in a container like the other workflows)

File: .github/workflows/release-tagged.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          for i in $(seq 0 $last); do
+            # Prefix the arch with linux_ if it's not windows
+            arch="${arches[i]}"
+
+            # Get the checksum for the arch we need
+            checksum=$(grep ${archive_names[i]} checksums.txt | awk '{print $1}')
+
+            # Update the checksum in the config
+            echo "Updating ${arch} checksum to ${checksum}"
+            sed -i "s/\b${arch}: .*/${arch}: ${checksum}/" stackbrew-config.yaml
```
Comment (hairyhenderson) [inline .github/workflows/release-tagged.yml:62]: this should work with `yq`:

```suggestion
            yq write -i stackbrew-config.yaml ${arch} ${checksum}
```

File: .github/workflows/release-tagged.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+          # Get the last index for the loop
+          last=$(expr "${#arches[@]}" - 1)
+
+          for i in $(seq 0 $last); do
+            # Prefix the arch with linux_ if it's not windows
+            arch="${arches[i]}"
+
+            # Get the checksum for the arch we need
+            checksum=$(grep ${archive_names[i]} checksums.txt | awk '{print $1}')
```
Comment (hairyhenderson) [inline .github/workflows/release-tagged.yml:58]: don't need `grep`:

```suggestion
            checksum=$(awk "/${archive_names[i]}/{print $1}")
```

**Comment by francislavoie** [comment]
Hmm. I'm not liking how opinionated `yq` is about whitespace.

I think I'll hold off on using `yq` for now because they have those whitespace changes tracked as a bug: https://github.com/mikefarah/yq/issues/343

**Comment by francislavoie** [comment]
@hairyhenderson alright what do you think is left here before we start testing the PR thing? I'll probably need to reopen this PR on a branch on this repo instead of on my fork to test that out... unless I try to make PRs against my fork? idk.

**Comment by hairyhenderson** [comment]
> Maybe this job only deals with the stackbrew conf and the other job updates the git refs? What's your plan there?

That makes the most sense. Ultimately, the stackbrew file (`library/caddy`) needs to be PRed in the official-images repo.

I think the best approach for now is that when a release is tagged in `caddyserver/caddy`, it triggers a job in this repo which updates the version and hashes in `stackbrew-config.yaml`, and updates the `Dockerfile`s appropriately (i.e. `make gen-dockerfiles`). For `library/caddy` to be properly generated, the commit must exist in `master` anyway, so that can be a thing that happens later.

File: .github/workflows/release-tagged.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        uses: actions/checkout@v2
+
+      # We disabled yq for now because of an upstream bug with whitespace
+      # See https://github.com/mikefarah/yq/issues/343
+      # - name: Install yq
+      #   run: |
+      #     sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CC86BB64
+      #     sudo add-apt-repository ppa:rmescandon/yq
+      #     sudo apt update
+      #     sudo apt install yq -y
```
Comment (hairyhenderson) [inline .github/workflows/release-tagged.yml:28]: If `yq` is needed (later?) it should probably just come from the `hairyhenderson/dockerfiles-builder` image... FWIW, I've added it in https://github.com/hairyhenderson/dockerfiles/commit/ad2d2b672ee4eda14fe6cb92c40b87f5d1f666e1

**Comment by hairyhenderson** [comment]
> I'll probably need to reopen this PR on a branch on this repo instead of on my fork to test that out

Yeah, best to branch from this repo, we'll have permission issues otherwise.

File: .github/workflows/release-tagged.yml (hunk 4 above)
Comment (francislavoie) [reply .github/workflows/release-tagged.yml:28]: Cool - yeah I'll just remove the yq stuff for now. We can revisit it later if we feel the need to (but I think sed is perfectly fine here :man_shrugging:)
